Assets/Scripts/Layers/Properties/PropertyPanels/TransformPropertySection.cs

```diff
@@ -122,57 +133,127 @@ private void SetTransformLocks()
+            
+            position.xField.SetTextWithoutNotify($"{rdCoordinate.Points[0].ToString(format, CultureInfo.InvariantCulture)}{positionUnitCharacter}");
+            position.yField.SetTextWithoutNotify($"{rdCoordinate.Points[1].ToString(format, CultureInfo.InvariantCulture)}{positionUnitCharacter}");
+            position.zField.SetTextWithoutNotify($"{rdCoordinate.Points[2].ToString(format, CultureInfo.InvariantCulture)}{positionUnitCharacter}");
         }
 
         private void UpdateRotationFields(Vector3 eulerAngles)
         {
-            rotation.xField.SetTextWithoutNotify(eulerAngles.x.ToString("0.00", CultureInfo.InvariantCulture));
-            rotation.yField.SetTextWithoutNotify(eulerAngles.y.ToString("0.00", CultureInfo.InvariantCulture));
-            rotation.zField.SetTextWithoutNotify(eulerAngles.z.ToString("0.00", CultureInfo.InvariantCulture));
+            var format = GetFormatString(rotationDecimals);
+            
+            rotation.xField.SetTextWithoutNotify($"{eulerAngles.x.ToString(format, CultureInfo.InvariantCulture)}{rotationUnitCharacter}");
+            rotation.yField.SetTextWithoutNotify($"{eulerAngles.y.ToString(format, CultureInfo.InvariantCulture)}{rotationUnitCharacter}");
+            rotation.zField.SetTextWithoutNotify($"{eulerAngles.z.ToString(format, CultureInfo.InvariantCulture)}{rotationUnitCharacter}");
         }
 
         private void UpdateScalingFields(Vector3 localScale)
         {
-            var xPercentage = localScale.x * 100;
-            var yPercentage = localScale.y * 100;
-            var zPercentage = localScale.z * 100;
+            var xPercentage = localScale.x * scaleMultiplier;
+            var yPercentage = localScale.y * scaleMultiplier;
+            var zPercentage = localScale.z * scaleMultiplier;
+            var format = GetFormatString(scaleDecimals);
+
+            scale.xField.SetTextWithoutNotify($"{xPercentage.ToString(format, CultureInfo.InvariantCulture)}{scaleUnitCharacter}");
+            scale.yField.SetTextWithoutNotify($"{yPercentage.ToString(format, CultureInfo.InvariantCulture)}{scaleUnitCharacter}");
+            scale.zField.SetTextWithoutNotify($"{zPercentage.ToString(format, CultureInfo.InvariantCulture)}{scaleUnitCharacter}");
+        }
+
+        public static string GetFormatString(int decimals)
+        {
+            if (decimals == 0)
+                return "0";
+
+            string format = "0.";
+            for (int i = 0; i < decimals; i++)
+            {
+                format += '0';
+            }
 
-            scale.xField.SetTextWithoutNotify($"{xPercentage.ToString("0", CultureInfo.InvariantCulture)}{percentageCharacter}");
-            scale.yField.SetTextWithoutNotify($"{yPercentage.ToString("0", CultureInfo.InvariantCulture)}{percentageCharacter}");
-            scale.zField.SetTextWithoutNotify($"{zPercentage.ToString("0", CultureInfo.InvariantCulture)}{percentageCharacter}");
+            return format;
```

Comment:
string defaultZero = "0";
string defaultZeroSeperator = "0.";
en
format += defaultZero;

nog beter een stringbuilder maar dat is in dit geval overkill